Sources/SentrySwiftUI/Preview/SentryReplayMaskPreview.swift

```diff
@@ -0,0 +1,43 @@
+#if canImport(SwiftUI) && canImport(UIKit) && os(iOS) || os(tvOS)
+import Sentry
+import SwiftUI
+import UIKit
+
+#if CARTHAGE || SWIFT_PACKAGE
+@_implementationOnly import SentryInternal
+#endif
+
+@available(iOS 13, macOS 10.15, tvOS 13, *)
+struct SentryReplayMaskPreview: ViewModifier {
+    let redactOptions: SentryRedactOptions
+    let opacity: Float
+    func body(content: Content) -> some View {
+        content.overlay(SentryReplayPreviewView(redactOptions: redactOptions, opacity: opacity))
+    }
+}
+
+@available(iOS 13, macOS 10.15, tvOS 13, *)
+public extension View {
+    func sentryReplayPreviewMask(redactOptions: SentryRedactOptions? = nil, opacity: Float = 1) -> some View {
+        let options = redactOptions ?? SentrySDK.options?.sessionReplay ?? PreviewRedactOptions()
+        return modifier(SentryReplayMaskPreview(redactOptions: options, opacity: opacity))
+    }
+}
+
+@available(iOS 13, macOS 10.15, tvOS 13, *)
+struct SentryReplayPreviewView: UIViewRepresentable {
+    let redactOptions: SentryRedactOptions
+    let opacity: Float
+    
+    func makeUIView(context: Context) -> UIView {
+        let view = SentryReplayMaskPreviewUIView(redactOptions: redactOptions)
+        view.isUserInteractionEnabled = false
```

Comment:
`m`: Did you test this with older iOS versions, that it does not block interactive elements under the overlay? We might have to implement a hit-test proxy view.

Reply:
So weird. I have being using `isUserInteractionEnabled` since forever to ignore hits. It was never a problem to me, but it looks like its a problem for SwiftUI.